kategory/src/main/kotlin/kategory/data/ListKW.kt

```diff
@@ -0,0 +1,56 @@
+package kategory
+
+typealias ListKindW<A> = HK<ListKW.F, A>
+
+fun <A> ListKindW<A>.ev(): ListKW<A> =
+        this as ListKW<A>
+
+data class ListKW<A> constructor(val list: List<A>) : ListKindW<A>, Collection<A> by list {
```

Comment:
I initially thought it was a good idea to do `Collection<A> by list` but after using this implementation I realized all the delegated combinators don't return a `ListKW` but a `List`.
so if we use `map` we get a `ListKW` but if we use `mapIndexed` we get back a `List` which is not consistent. Either we remove this delegation or we add manual delegation to all the methods in List and wrap their results in `ListKW`